cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/services/AbstractRunFixerService.java

```diff
@@ -213,10 +225,34 @@ private boolean shouldCorrectForWorkflowChildren(RunRecordMeta runRecordMeta,
   @Override
   protected void startUp() throws Exception {
     LOG.info("Starting RunRecordCorrectorService");
+    localDatasetDeleterService = Executors.newScheduledThreadPool(1);
```

Comment:
So changes to this class is just to not having a separate Service class to start the local dataset deleter thread?? Maybe it should go to `DatasetService`??